File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int x=0,n=nums.size();
        for(int i=0; i<n; i++){
            if(nums[i]!=val){
                nums[x]=nums[i];
                x++;
            }
        }
        return x;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        curr = []
        nums.sort()
        l = len(nums)
        def dfs(i):
            if i == l:
                res.append(curr.copy())
                return
            curr.append(nums[i])
            dfs(i+1)

            curr.pop()
            while i<=l-2 and nums[i]==nums[i+1]:
                i+=1
            dfs(i+1)

        dfs(0)
        return res
            
            